File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int[] arr = new int[nums.length+1];
        for (int i = 0;i<nums.length;i++)
        {
            arr[i] = nums[i];
        }
        arr[nums.length] = 0;
        for (int i = arr.length-3;i>-1;i--)
        {
            arr[i] = Math.max(arr[i]+arr[i+2], arr[i+1]);
        }
        return arr[0];
    }
}

File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return nums[0]
        if len(nums) == 2:
            return max(nums[0], nums[1])
        arr = [0] * len(nums)
        arr[0] = nums[0]
        arr[1] = nums[1]
        for i in range(2, len(nums)):
            maxval = 0
            for j in range(0, i-1):
                maxval = max(maxval, arr[j])
            arr[i] = nums[i] + maxval
        
        return max(arr[-1], arr[-2])